vpr/src/base/vpr_constraints_serializer.h

```diff
@@ -128,8 +153,10 @@ class VprConstraintsSerializer final : public uxsd::VprConstraintsBase<VprConstr
     virtual inline void finish_partition_add_atom(void*& /*ctx*/) final {
         PartitionId part_id(num_partitions_);
 
-        if (atom_id_ != AtomBlockId::INVALID()) {
-            constraints_.add_constrained_atom(atom_id_, part_id);
+        for (unsigned int i = 0; i < atoms_.size(); i++) {
+            if (atoms_[i] != AtomBlockId::INVALID()) {
```

Comment:
shouldn't need if statement because this should never happen, get rid of the if statement. can use vtr assert -> blockid is not invalid. if atom netlist has an obvious check that block id is valid, can use that here. if there is no easy check, don't need to do that